cluster/node.go

```diff
@@ -0,0 +1,72 @@
+package cluster
+
+import (
+	"sync"
+	"time"
+)
+
+type NodeState int
+
+const (
+	NodeReady       NodeState = 0
+	NodeNotReady    NodeState = 1
+	NodeUnreachable NodeState = 2
```

Comment:
for future cluster support when nodes have peers that could be in an unreachable state.